include/zephyr/drivers/emul.h

```diff
@@ -135,6 +144,49 @@ extern const struct emul __emul_list_end[];
 				&(Z_EMUL_REG_BUS_IDENTIFIER(node_id))},                            \
 	};
 
+#define Z_MAYBE_EMUL_DECLARE_INTERNAL(node_id) extern const struct emul EMUL_DT_NAME_GET(node_id);
+
+DT_FOREACH_STATUS_OKAY_NODE(Z_MAYBE_EMUL_DECLARE_INTERNAL)
+
+/**
+ * @brief Define a new emulator
+ *
+ * Prefer EMUL_DT_DEFINE
+ */
+#define EMUL_DEFINE(init_ptr, node_id, cfg_ptr, data_ptr, bus_api)                                 \
```

Comment:
I think we should just remove `EMUL_DEFINE`.

Reply:
Added a commit to remove the old names. (I think I had removed this one before, and it snuck back in during a rebase.)